cypress/component/DataSearch/dataset_search_table.spec.js

```diff
@@ -38,6 +43,47 @@ describe('Dataset Search Table tests', () => {
       cy.get('#header-checkbox').click();
       cy.contains('1 dataset selected from 1 study');
     });
+  });
+
+  describe('Data library filter by participant count tests', () => {
+    let searchText;
+    let filtered;
+
+    beforeEach(() => {
+      cy.initApplicationConfig();
+      cy.stub(TerraDataRepo, 'listSnapshotsByDatasetIds').returns({});
+      filtered = false;
+    });
+
+    function handler(request) {
+      if (JSON.stringify(request.body).includes(searchText)) {
+        filtered = true;
+      }
+      request.reply({statusCode: 200, body:[]});
+    }
+
+    it('When a participant count filter is applied the query is updated', () => {
+      searchText ='{"range":{"participantCount":{"gte":null,"lte":"50"}}}';
+
+      cy.intercept(
+        {method: 'POST', url: '**/api/dataset/search/index'}, handler);
+      mount(<DatasetSearchTable {...props} />);
+      cy.get('#participantCountMax-range-input').clear().type('50');
+      cy.wait(3000).then(() => {
+        expect(filtered).to.be.true;
+      });
```

Comment:
The format of the intercept + wait approach should look something like this:
```
      cy.intercept(
        {method: 'POST', url: '**/api/dataset/search/index'}, handler).as('searchIndex');
      ...
      cy.wait('@searchIndex').then(() => {
        expect(filtered).to.be.true;
      });
```
We should avoid using timeouts in wait calls.